### DISCUSSION TITLE
Documentation/code sample > `.gomplate.yaml` example

### DISCUSSION THREAD

**Discussion by adriens** [hairyhenderson/gomplate#1105] [Q&A] [answered]
Hi, I currently have the following command line that works : 

`gomplate -d data=person.json --file template.tmplt --out OUT.txt`

But when I try to convert it to a `.gomplate.yaml` , I cannot achieve to define the output file, only to get to the console.

Could you please give a sample that shows how to use `file` and `out` paramaters in `.gomplate.yaml`  ?

Thank you in advance for any help.

**Comment by hairyhenderson** [comment, marked as answer]
Hi @adriens - I've moved this to the Discussions section because this is more of a question than an issue 😉

One way to generate a config file from a given commandline is to run gomplate in [verbose mode](https://docs.gomplate.ca/usage/#verbose). It'll print out the YAML form of the configuration that it's about to run. Here's what I get on my system with the same commandline:

```yaml
$ gomplate -V -d data=person.json --file template.tmplt --out OUT.txt
17:01:58 DBG starting gomplate
17:01:58 DBG config is:
---
inputFiles: [template.tmplt]
outputFiles: [OUT.txt]
leftDelim: '{{'
rightDelim: '}}'
datasources:
  data:
    url: file:///Users/hairyhenderson/person.json
    header: {}
pluginTimeout: 5s
 build= version=3.9.0
17:01:58 DBG created data from config data="&{ctx:0xc0000aedb0 Sources:map[data:data=file:///Users/hairyhenderson/person.json ()] sourceReaders:map[] cache:map[] extraHeaders:map[]}"
17:01:58 DBG completed rendering duration=0.000340407 errors=0 templatesRendered=1
```

So, here's a configuration that would work:

```yaml
---
inputFiles: [template.tmplt]
outputFiles: [OUT.txt]
leftDelim: '{{'
rightDelim: '}}'
datasources:
  data:
    url: file:///Users/hairyhenderson/person.json
    header: {}
pluginTimeout: 5s
```

Some of these values are unnecessary since they're defaults, and the formatting isn't ideal. Here's an equivalent, but more readable version:

```yaml
inputFiles:
  - template.tmplt

outputFiles:
  - OUT.txt

datasources:
  data:
    url: person.json
```

Note: I've changed the `url` property to be a relative path, which is supported only for file datasources.

Hope this explains things for you! Let me know if you have any more questions.

**Comment by adriens** [comment]
Hi @hairyhenderson , sorry for late answer... my turn to have a rought week ;-(

Thanks a lot for your answer, yes, it perfeclty matches my needs and answers my question.

By the way, generatin the file conf in verbose mode is a great idea, I really 💌  the approach.

Thanks a lot for your kind answer.

Kind Regards,
Adrien

**Reply by hairyhenderson** [reply]
@adriens if that answered the question can you mark it as an answer? Just click the icon in the upper right of the response

<img width="161" alt="Screen Shot 2021-04-01 at 20 03 18" src="https://user-images.githubusercontent.com/2037086/113365802-4f64a100-9325-11eb-9df4-2c9230da5cbb.png">

**Comment by adriens** [comment]
Yes, sorry, not yet the habit to deal with GH discussions 😸
